src/xtgeo/interfaces/rms/rmsapi_utils.py

```diff
@@ -0,0 +1,442 @@
+
+        Returns:
+            A list of RMS version(s) for the given API version, None if
+                not any match.
+
+        Example::
+
+            rox = RmsApiUtils(project)
+            rmsver = rox.rmsversion('1.5')
+            print('The supported RMS version are {}'.format(rmsver))
+
+        """
+
+        return self._versions.get(apiversion, None)
+
+    def _create_whatever_category(
+        self,
+        category: str | list[str],
+        stype: str = "horizons",
+        domain: Literal["depth", "time", "thickness", "unknown"] = "depth",
+        htype: Literal["surface", "lines", "points"] = "surface",
+    ) -> None:
+        """Create one or more a Horizons/Zones... category entries.
+
+        Args:
+            category (str or list): Name(s) of category to make, either as
+                a simple string or a list of strings.
+            stype (str): 'Super type' in RMS (horizons or zones).
+                Default is horizons
+            domain (str): The vertical_domian in RMS, 'depth' (default) or 'time'
+            htype (str): Horizon type: surface/lines/points
+        """
+
+        project = self.project
+        # At this point, project is always a Project instance
+        # (str/Path converted in __init__)
+        assert rmsapi is not None
+        assert isinstance(project, rmsapi.Project), "Project must be initialized"
+
+        categories = []
+
+        if isinstance(category, str):
+            categories.append(category)
+        else:
+            categories.extend(category)
+
+        for catg in categories:
+            geom = rmsapi.GeometryType.surface
+            if htype.lower() == "lines":
+                geom = rmsapi.GeometryType.polylines
+            elif htype.lower() == "points":
+                geom = rmsapi.GeometryType.points
+
+            dom = rmsapi.VerticalDomain.depth
+            if domain.lower() == "time":
+                dom = rmsapi.VerticalDomain.time
+            elif domain.lower() == "thickness":
+                dom = rmsapi.VerticalDomain.thickness
+            elif domain.lower() == "unknown":
+                dom = rmsapi.VerticalDomain.unknown
+
+            if stype.lower() == "horizons":
+                if catg not in project.horizons.representations:
+                    try:
+                        project.horizons.representations.create(catg, geom, dom)
+                    except Exception as exmsg:
+                        print(f"Error: {exmsg}")
+                else:
+                    print(f"Category <{catg}> already exists")
+
+            elif stype.lower() == "zones":
+                if catg not in project.zones.representations:
+                    try:
+                        project.zones.representations.create(catg, geom, dom)
+                    except Exception as exmsg:
+                        print(f"Error: {exmsg}")
+                else:
+                    print(f"Category <{catg}> already exists")
+
```

Comment:
Might be me that does not have the full picture, but shouldn't there be operations for creating clipboard and general_2d folders as well or is this handled elsewhere? 🙂